std/experimental/allocator/package.d

```diff
@@ -1600,6 +1609,67 @@ unittest //bugzilla 15721
     Mallocator.instance.dispose(foo);
 }
 
+unittest //bugzilla 16824
+{
+    struct TestAllocator
+    {
+        import std.experimental.allocator.common: platformAlignment;
+        import std.experimental.allocator.mallocator: Mallocator;
+
+        alias allocator = Mallocator.instance;
+
+        private static struct ByteRange
+        {
+            void* ptr;
+            size_t length;
+        }
+
+        private ByteRange[] _allocations;
+
+        enum uint alignment = platformAlignment;
+
+        void[] allocate(size_t numBytes)
+        {
+             auto ret = allocator.allocate(numBytes);
+             _allocations ~= ByteRange(ret.ptr, ret.length);
+             return ret;
+        }
+
+        bool deallocate(void[] bytes)
+        {
+            import std.algorithm: remove, canFind;
+            import std.conv: text;
+
+            bool pred(ByteRange other)
+            { return other.ptr == bytes.ptr && other.length == bytes.length; }
+
+            assert(_allocations.canFind!pred);
+
+             _allocations = _allocations.remove!pred;
```

Comment:
FYI: there is `countUntil` which allows to the search only once, but of course it doesn't matter for this simple test.